CMakeLists.txt

```diff
@@ -81,6 +85,10 @@ else()
     option(BUILD_SHARED_LIBS "Build shared libraries" ON)
 endif()
 
+if(BUILD_SHARED_LIBS AND MINGW)
+    add_definitions(-D_DLL)
```

Comment:
Define `_DDL` use almost all packages and dependencies. Do you offer to define it for every target?

Reply:
You should ask your self, if it is mandatory to set also _DLL into your project they use the compiled poco library ? Mostly I checked in the code if this preprocessor definiton is in the header file .... if not, then it can be `PRIVATE` (it is only need to compile the poco librarie) ... if it is in the header file, then you should investigate is this header file published with the compiled library ... if yes then it is mostly mandatory that you added this as `PUBLIC` because it is needed to use the compiled poco library into your project. You should not added these to every target. Added with `target_compile_definition` with `PUBLIC` there where it is used and this definiton will be provided to the other targets they use this target. For example for me we should add this on the Poco::Foundation target. All of the other targets depends on Poco::Foundation and get this definiton ...See also [target_compile_definitions](https://cmake.org/cmake/help/latest/command/target_compile_definitions.html?highlight=target_compile_#command:target_compile_definitions) I found that _DLL is used in Crypto.h, CppUnit.h, Foundation,h, NetSSL.h, pngconf.h .... I see also we have POCO_DLL ... hmmm is this the same?